core/base/v7/inc/ROOT/TDirectory.h

```diff
@@ -96,47 +92,51 @@ class TDirectory {
   /// \param name  - Key of the object.
   /// \param args  - arguments to be passed to the constructor of `T`
   template <class T, class... ARGS>
-  std::weak_ptr<T> Create(const std::string& name, ARGS... args) {
-    auto ptr = std::make_shared<ToContentType_t<T>>(std::forward<ARGS...>(args...));
+  decltype(auto) Create(const std::string& name, ARGS&&... args) {
```

Comment:
I'd prefer to spell out the return type, for three reasons:
- we can :-)
- it's one of the main user interfaces; we should be specific here.
- the error message becomes nicer when calling `Create<T>` with a `T` that is enable_if-ed away by `ToContentType`.
Could you make that change?